test/index.js

```diff
@@ -1,257 +1,415 @@
+  } catch (e) {
+    throw new Error(e);
+  }
+};
 
-      expect(spawnStub.calledWith(sinon.match.string, expected)).to.be.true;
-    });
+beforeEach(setup);
 
-    it('should pass canary channel to "version chrome"', async () => {
-      const args = ['install', 'chromedriver=canary'];
-      const expected = ['version', 'chrome=canary'];
-      mockStdoutChunk = 'Google Chrome 97.0.4692.71';
-      console.log = mockConsoleLog;
-      await browserDriverManager(args);
+afterEach(teardown);
 
-      expect(spawnStub.calledWith(sinon.match.string, expected)).to.be.true;
-    });
+const makeEnvFile = async (contents = envContents) => {
+  await fsPromises.mkdir(MOCK_BDM_CACHE_DIR, { recursive: true });
+  await fsPromises.writeFile(envPath, contents);
+};
 
-    it('should error if chromedriver version is invalid', async () => {
-      const args = ['install', 'chromedriver=invalid'];
-      console.log = mockConsoleLog;
-      console.error = mockConsoleError;
-      await browserDriverManager(args);
+describe('browser-driver-manager', () => {
+  describe('which', async () => {
+    it('logs the locations of chrome and chromedriver if they exist', async () => {
+      await makeEnvFile();
+      await which();
 
-      expect(mockConsoleError.calledWith(sinon.match.string, 'Chrome version "invalid" is not a number')).to.be.true;
+      sinon.assert.calledWith(mockConsoleLog, envContents);
     });
-
-    it('should correctly get major version of chrome', async () => {
-      const args = ['install', 'chromedriver'];
-      mockStdoutChunk = 'Google Chrome 97.0.4692.71';
-      console.log = mockConsoleLog;
-      await browserDriverManager(args);
-
-      expect(mockConsoleLog.calledWith('Installing ChromeDriver 97')).to.be.true;
+    it('errors if no environment file exists', async () => {
+      await expect(which()).to.be.rejectedWith(
+        'No environment file exists. Please install first'
+      );
     });
+  });
+  describe('version', () => {
+    it('logs the version when a valid one exists', async () => {
+      await makeEnvFile();
+      await version();
 
-    it('should fetch npm chromedriver version', async () => {
-      const args = ['install', 'chromedriver', '--verbose'];
-      mockStdoutChunk = 'Google Chrome 97.0.4692.71';
-      console.log = mockConsoleLog;
-      await browserDriverManager(args);
-
-      expect(chromdriverNock.isDone()).to.be.true;
+      sinon.assert.calledWith(mockConsoleLog, mockVersion);
     });
-
-    it('should use the version if it is equal to the latest npm chromedriver', async () => {
-      const args = ['install', 'chromedriver=97'];
-      const expected = ['install', '--no-save', 'chromedriver@97'];
-      console.log = mockConsoleLog;
-      await browserDriverManager(args);
-
-      expect(spawnStub.calledWith('npm', expected)).to.be.true;
+    it('errors if no environment file exists', async () => {
+      await expect(which()).to.be.rejectedWith(
```

Comment:
Test shouldn't pass as `version` doesn't currently error.

```suggestion
      await expect(version()).to.be.rejectedWith(
```